Clarifications between app and bootloader binaries

bridadan opened the pull request:
**NOTE:** I'm currently submitting this against the `mbed-os-5.5.0-rc1-oob` branch since that's what I was testing. If I should submit this against `master` instead please let me know.

This attempts to clarify a few vague points in the README. It helps to
distinguish between the different binaries created during the bootloader
building process.

I also moved a few parenthesized phrases ([you're darn right that's a real word](https://www.merriam-webster.com/dictionary/parenthesize)) to the ends of sentences. When I read them they broke up the sentence too much and made them hard to understand.

@c1728p9 Can you make sure I didn't mess up the jargon here? Specifically, I swapped the word "combined" with "padded/unpadded" when building the bootloader binaries. I think that's right, though not 100% sure.

@AnotherButler A general English sanity check would be much appreciated 😄 And probably my use of the `tips` tag.

bridadan commented:
Huh, just realized this is mostly the same as #10, lol oops. Well hopefully it's just a good iteration of it.

theotherjimmy commented on README.md:
````diff
@@ -74,15 +74,17 @@ Total Flash memory (text + data + misc): 60554 bytes
 ```
 
-It creates two binary files. The original uncombined image is in the output directory, <project-name>_application.bin, and the bootloader image is <project-name>.bin.
+It creates two binary images in the `BUILD/<TARGET_NAME>/<TOOLCHAIN_NAME>` directory. The original unpadded image is saved with the name `<project-name>_application.bin`. The padded bootloader image (used for combining with applications) is saved with the name `<project-name>.bin`.
````
What is "it" in this context?

Who or what saves this the images?

theotherjimmy commented on README.md:
```diff
@@ -74,15 +74,17 @@ Total Flash memory (text + data + misc): 60554 bytes
+It creates two binary images in the `BUILD/<TARGET_NAME>/<TOOLCHAIN_NAME>` directory. The original unpadded image is saved with the name `<project-name>_application.bin`. The padded bootloader image (used for combining with applications) is saved with the name `<project-name>.bin`.
 
-When the build succeeds, you have created a bootloader for your target. This bootloader defines the update file to be located at `/sd/mbed-os-example-bootloader-blinky_application.bin`. This is the binary bootloader flashes and then jumps to.
+When the build succeeds, you have created a bootloader for your target. This bootloader defines the update binary to be located at `/sd/mbed-os-example-bootloader-blinky_application.bin` on the sd card. This is the binary that the bootloader flashes and then jumps to.
```
Can we just remove this first sentence? Could we clarify that the sd card must be formatted as FAT? Could we reword the second sentence to: "This bootloader looks for the update binary at `/mbed-os-example-bootolader-blinky_application.bin` on the attached sd card formatted with the FAT filesystem. When this bootloader finds this update binary, it flashes the new application and jumps to it."

AnotherButler replied:
Also, please capitalize "SD".

theotherjimmy commented on README.md:
```diff
@@ -74,15 +74,17 @@ Total Flash memory (text + data + misc): 60554 bytes
-When the build succeeds, you have created a bootloader for your target. This bootloader defines the update file to be located at `/sd/mbed-os-example-bootloader-blinky_application.bin`. This is the binary bootloader flashes and then jumps to.
+When the build succeeds, you have created a bootloader for your target. This bootloader defines the update binary to be located at `/sd/mbed-os-example-bootloader-blinky_application.bin` on the sd card. This is the binary that the bootloader flashes and then jumps to.
+
+<span class="tips">**Note:** The `/sd/` part of the update binary path **does not** correspond to a directory of `sd` on the SD card's filesystem. The entire SD card is mounted to the filesystem as the directory `/sd/`. For this example, you should place the update binary in the root directory of the SD card.</span>
```
This is a good tip, but if this is the case, why did we include the `/sd` in the update binary path above?

bridadan replied:
> why did we include the /sd in the update binary path above?

Good question, I didn't write this thing.

AnotherButler replied:
- Please make "file system" two words both times.
- Query: Who or what is mounting the SD card to the file system? Please clarify.
- In the last sentence, please change "you should place" to "place".

bridadan replied:
I'm removing this whole bit, so that should address your comments.

theotherjimmy commented on README.md:
```diff
@@ -74,15 +74,17 @@ Total Flash memory (text + data + misc): 60554 bytes
 ## Set up application to use bootloader
 
-The next step is to build an application you can combine with your bootloader to create a loadable image. 
+The next step is to build an application you can combine with your bootloader to create a loadable image. Any application should work, however you may wish to start with the [blinky example](https://github.com/ARMmbed/mbed-os-example-blinky).
```
The first sentence is passive voice. Could you add a subject to it, and make the tense present?

AnotherButler replied:
Actually, the first sentence is active voice and present tense. In the second sentence, however: 

- Please change "should work" to "works".
- Please change "work, however you" to work; however, you".
- Please capitalize "Blinky".

theotherjimmy commented on README.md:
````diff
@@ -99,16 +101,20 @@ The next step is to build an application you can combine with your bootloader to
     mbed compile -m <TARGET_NAME> -t ARM
     ```
 
+It creates two binary images in the `BUILD/<TARGET_NAME>/<TOOLCHAIN_NAME>` directory. The image that contains the combined bootloader and application (suitable for flashing to the device's ROM directly) is saved with the name `<project-name.bin>`. The image that should be used as the update binary (the one that should be placed on the SD card) is saved with the name `<project-name>_application.bi ...
````
Who or what is "It" in this first sentence? Could you change "is saved with the name" to "is named" or better yet: "The image named `<project-name>.bin` contains the combined bootloader and application suitable for flashing to the device's ROM directly". Could you edit the third sentence in the same fashion as the second?

AnotherButler replied:
Specifically, please change the third sentence to "Use the image saved with the name `<project-name>_application.bin' as the update binary (the one you place on the SD card)."

theotherjimmy reviewed (commented):
I provided some edit suggestions below.

bridadan commented:
@theotherjimmy You managed to comment on pretty much every line I didn't change 😛 

Nevertheless, I'll update this.

AnotherButler commented on README.md:
```diff
@@ -74,15 +74,17 @@ Total Flash memory (text + data + misc): 60554 bytes
+The next step is to build an application you can combine with your bootloader to create a loadable image. Any application should work, however you may wish to start with the [blinky example](https://github.com/ARMmbed/mbed-os-example-blinky).
 
-1. Update your board to use the newly created bootloader image. To do this, set the target (replace `<TARGET_NAME>` with your target name) value `bootloader_img` to the file path of the bootloader image.
+1. Instruct the build system to use the newly created padded bootloader image (named `<project-name>.bin` in the previous section). To do this, set the target value `bootloader_img` to the file path of the bootloader image (replace `<TARGET_NAME>` with your target name). In the below example, the bootloader image has been copied to a new directory called `bootloader` in the project and renamed to ...
```
Query: Who or what has copied the bootloader image to the new directory and renamed it? Please clarify.

AnotherButler commented on README.md:
```diff
@@ -99,16 +101,20 @@ The next step is to build an application you can combine with your bootloader to
 
 ### Program application using SD card
 
+<span class="tips">**Note:** You must first flash the combined bootloader and application image or just the bootloader image (padded or unpadded) before proceeding with the following steps</span>
```
Please add a period after "steps" because this is a complete sentence.

AnotherButler reviewed (commented):
@bridadan Thanks so much for the great content. I left a few minor grammar comments. 

@theotherjimmy Thanks for making the initial grammar changes while I was out. Much appreciated.

bridadan committed:
Distinguish app and bootloader binaries. Use app config.

This attempts to clarify a few vague points in the README. It helps to
distinguish between the different binaries created during the bootloader
building process.

It takes care of the somewhat confusing '/sd/' mount path for the
SD card. This is now handled by the program, allowing the user to just
specify the name of the binary that is placed in the root of the SD card.

It makes more use of app config to support onboard SD card slots
for the K64F and the UBLOX_EVK_ODIN_W2. It will use the Arduino SPI pins
for any other target.

The application update binary name is now specified in the app config as
well.

bridadan commented:
This has definitely evolved from just a README update! I removed the ambiguity of the SD mount point in the app. Now we don't have to mention the `/sd/` path in the README. We can stick to just the filesystem of the SD card and not worry how the program mounts it.

@theotherjimmy @adbridge Could you take a look?

@c1728p9 You might wanna do a once over as well since I'm touching the program code.

c1728p9 commented:
Thanks for the doc updates @bridadan. It looks good

bridadan commented:
This PR has deviated significantly from the original intent and we already have a new RC for mbed OS 5.5.0. I'm gonna close this and reopen.
